marshal/claim.go

```diff
@@ -311,7 +314,16 @@ func (c *claim) messagePump() {
 		c.lock.Unlock()
 
 		// Push the message down to the client (this bypasses the Consumer)
-		c.messages <- msg
+		// We should NOT write to the consumer channel if the claim is no longer claimed. This
+		// needs to be serialized with Release, otherwise a race-condition can potentially
+		// lead to a write to a closed-channel. That's why we're using this lock. We're not
+		// using the main lock to avoid deadlocks since the write to the channel is blocking
+		// until someone consumes the message blocking all Commit operations
+		c.messagesLock.Lock()
```

Comment:
I'm pretty dubious about adding a second lock here. I'm trying to think of a way to handle this without locking, but nothing has immediately come to mind... I will think about it some more over the weekend and try to come up with something